### PR TITLE
Fix/psl

### LINKED ISSUE #169: Bug with `getEffectiveTLDPlusOne`
`getEffectiveTLDPlusOne` parses incorrectly when a TLD+2 domain may resemble but not exactly match the input.

Steps to repro:
```
getEffectiveTLDPlusOne('jurajuhlar.com')
```
Expected:
```
'jurajuhlar.com'
```
Actual:
```
''
```
Reason:
`'jurajuhlar.com'` gets matched with `ar.com` TLD.

### PR DESCRIPTION
**Pull request by sshelomentsev**

### CODE REVIEW

**Review by ilfa — DISMISSED** [review]

File: src/utils/getEffectiveTLDPlusOne.ts [hunk 1]
```diff
@@ -64,14 +68,93 @@ function createTrie(): TrieNode {
 function search(domain: string): string | null {
   let node = trie
 
-  for (let i = 0; i < domain.length; i++) {
+  let i = 0
```
Comment (necipallef) [inline src/utils/getEffectiveTLDPlusOne.ts]:
```suggestion
  let i
```

File: src/utils/getEffectiveTLDPlusOne.ts [hunk 2]
```diff
@@ -64,14 +68,93 @@ function createTrie(): TrieNode {
...
+  let currNode = node
+  while (currNode.parent != null) {
+    if (compare(currNode.suffix, sub)) {
+      return currNode.suffix
+    } else {
+      currNode = currNode.parent
+    }
+  }
+
+  if (currNode != null) {
```
Comment (necipallef) [inline src/utils/getEffectiveTLDPlusOne.ts]: Can `currNode` ever be `null`? Seems like no

```suggestion
  return currNode.suffix
```

File: src/utils/getEffectiveTLDPlusOne.ts [hunk 3]
```diff
@@ -64,14 +68,93 @@ function createTrie(): TrieNode {
...
+  }
+
+  if (v1.length === v2.length && v1.length === i) {
+    return true
+  }
+
+  const s1 = v1.substring(i)
+  const s2 = v2.substring(i)
+
+  if (s1 === '.*' || s1 === '*' || s2 === '.*' || s2 === '*') {
```
Comment (necipallef) [inline src/utils/getEffectiveTLDPlusOne.ts]:
```suggestion
  return s1 === '.*' || s1 === '*' || s2 === '.*' || s2 === '*'
```

File: test/utils/getEffectiveTLDPlusOne.test.ts [hunk 4]
```diff
@@ -23,6 +23,19 @@ const cases = [
   ['b.example.uk.com', 'example.uk.com'],
   ['a.b.example.uk.com', 'example.uk.com'],
   ['test.ac', 'test.ac'],
+  // TLD with 1 rule that may resemble a 2-level rule
+  ['b.a-uk.uk.com', 'a-uk.uk.com'],
+  ['a-uk.uk.com', 'a-uk.uk.com'],
+  ['a-uk.com', 'a-uk.com'],
+  ['lar.com', 'lar.com'], // should use com, but not ar.com
```
Comment (necipallef) [inline test/utils/getEffectiveTLDPlusOne.test.ts]:
```suggestion
  ['bar.com', 'bar.com'], // should use com, but not ar.com
```

File: test/utils/getEffectiveTLDPlusOne.test.ts [hunk 5]
```diff
@@ -23,6 +23,19 @@ const cases = [
...
+  ['dsfdsaar.com', 'dsfdsaar.com'],
+  ['www.lar.com', 'lar.com'],
```
Comment (necipallef) [inline test/utils/getEffectiveTLDPlusOne.test.ts]:
```suggestion
  ['www.bar.com', 'bar.com'],
```

File: test/utils/getEffectiveTLDPlusOne.test.ts [hunk 6]
```diff
@@ -23,6 +23,19 @@ const cases = [
...
+  ['seed4u.com', 'seed4u.com'],
```
Comment (necipallef) [inline test/utils/getEffectiveTLDPlusOne.test.ts]:
```suggestion
  ['foo4u.com', 'foo4u.com'],
```

File: test/utils/getEffectiveTLDPlusOne.test.ts [hunk 7]
```diff
@@ -23,6 +23,19 @@ const cases = [
...
+  ['lasthome.co.uk', 'lasthome.co.uk'], // should use co.uk, but not wellbeingzone.co.uk
```
Comment (necipallef) [inline test/utils/getEffectiveTLDPlusOne.test.ts]:
```suggestion
  ['foo.co.uk', 'foo.co.uk'], // should use co.uk, but not wellbeingzone.co.uk
```

File: test/utils/getEffectiveTLDPlusOne.test.ts [hunk 8]
```diff
@@ -23,6 +23,19 @@ const cases = [
...
+  ['ecaytrade.com', 'ecaytrade.com'],
```
Comment (necipallef) [inline test/utils/getEffectiveTLDPlusOne.test.ts]:
```suggestion
  ['foobarde.com', 'foobarde.com'],
```

File: test/utils/getEffectiveTLDPlusOne.test.ts [hunk 9]
```diff
@@ -23,6 +23,19 @@ const cases = [
...
+  ['shavekit.com', 'shavekit.com'],
```
Comment (necipallef) [inline test/utils/getEffectiveTLDPlusOne.test.ts]:
```suggestion
  ['foobarit.com', 'foobarit.com'],
```

File: test/utils/getEffectiveTLDPlusOne.test.ts [hunk 10]
```diff
@@ -23,6 +23,19 @@ const cases = [
...
+  ['axonet.eu.org', 'axonet.eu.org'],
```
Comment (necipallef) [inline test/utils/getEffectiveTLDPlusOne.test.ts]:
```suggestion
  ['foobar.eu.org', 'foobar.eu.org'],
```

File: test/utils/getEffectiveTLDPlusOne.test.ts [hunk 11]
```diff
@@ -23,6 +23,19 @@ const cases = [
...
+  ['cargambit.com', 'cargambit.com'],
```
Comment (necipallef) [inline test/utils/getEffectiveTLDPlusOne.test.ts]:
```suggestion
  
```

**Review by necipallef — APPROVED** [review]

**Pull request merged**